File: aoc-2015/src/test/java/com/capital7software/aoc2015/days/Day02Test.java
package com.capital7software.aoc2015.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.aoc2015.days.Day02;
import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day02Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day02Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day02();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void getTotalWrappingPaperNeeded() {
    var instance = new Day02();
    var actual = instance.howMuchTotalWrappingPaper(instance.loadPresents(lines));

    assertEquals(101, actual);
  }

  @Test
  void getTotalRibbonNeeded() {
    var instance = new Day02();
    var actual = instance.howMuchTotalRibbon(instance.loadPresents(lines));

    assertEquals(48, actual);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2016/src/test/kotlin/com/capital7software/aoc/aoc2016aoc/days/Day02Test.kt
package com.capital7software.aoc.aoc2016aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day02Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day02Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day02()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testCalculateSimpleBathroomCode() {
    val instance = Day02()
    val expected = "1985"
    val actual = instance.calculateSimpleBathroomCode(lines)

    assertEquals(expected, actual, "The room code is not what was expected: $expected")
  }

  @Test
  fun testCalculateComplexBathroomCode() {
    val instance = Day02()
    val expected = "5DB3"
    val actual = instance.calculateComplexBathroomCode(lines)

    assertEquals(expected, actual, "The room code is not what was expected: $expected")
  }
}
